src/goto-programs/goto_convert_side_effect.cpp

```diff
@@ -44,13 +44,14 @@ void goto_convertt::remove_assignment(
 
   if(statement==ID_assign)
   {
-    exprt new_lhs = skip_typecast(expr.op0());
+    auto &old_assignment = to_side_effect_expr_assign(expr);
```

Comment:
This is unnecessary, because `expr` already is a `side_effect_exprt`. But we could make `old_assignment` a const reference.

Reply:
added the const -- we still need it to be a `side_effect_expr_assignt`, or we don't have the .lhs() and .rhs()